MyRangesImplementation/MyRangesImplementation/cartezian_product_iterator.hpp

```diff
@@ -0,0 +1,112 @@
+#pragma once
+#include<type_traits>
+#include"range_traits.hpp"
+
+namespace ranges {
+	namespace internals {
+		namespace iterators {
+			template<typename... Iterators>
+			class cartezian_product_iterator {
+				static_assert(RangeTraits::are_all_iterators<Iterators...>::value,
+					"error: all template parameters should be iterators" );
+
+				std::tuple<std::tuple<Iterators, Iterators, Iterators>...> m_beginCurrenEndIterTuples;
```

Comment:
this member responsibility is not clear from its name

consider changing its name or adding a comment.
See JavaDoc Standard for comments (or any other standard you like that supports Doxygen)

Reply:
A using type declaration may also aid in readability